### PR TITLE
Refactor usage of args (excluding *args) to argv

### PR DESCRIPTION
**Pull request by itsjayway**
#28

Please advise if changes in https://github.com/intel/code-base-investigator/compare/main...itsjayway:code-base-investigator:main#diff-1b77d714cd5303809e8b061199c6146dc9dc66bb03443f8e27a25598b2411143R56-R61

or
https://github.com/intel/code-base-investigator/compare/main...itsjayway:code-base-investigator:main#diff-971c35bb84781bea01e028040afd4f1df63231ceca66defde899193a2375db15R99

are necessary, or if https://github.com/intel/code-base-investigator/pull/33/commits/5229969c6bc48f510165f3b016b85dd6bfccb59f was sufficient.

### CODE REVIEW

File: codebasin.py [hunk 1]
```diff
@@ -80,19 +80,19 @@ def guess_project_name(config_path):
                         help='dump out annotated platform/parsing tree to <file.json>')
     parser.add_argument('--batchmode', dest='batchmode', action='store_true', default=False,
                         help="Set batch mode (additional output for bulk operation.)")
-    args = parser.parse_args()
+    argv = parser.parse_args()
```
Comment (Pennycook) [inline codebasin.py:83]: I think this one should remain `args`:
```python
args = parser.parse_args()
```
... is the idiomatic way of writing this, and what `parse_args()` returns is not a list of argument strings.

(This comment applies to everything in codebasin.py.)

File: codebasin/preprocessor.py [hunk 2]
```diff
@@ -697,16 +697,16 @@ class DefineNode(DirectiveNode):
     A DirectiveNode representing a #define directive.
     """
 
-    def __init__(self, identifier, args=None, value=None):
+    def __init__(self, identifier, argv=None, value=None):
```
Comment (Pennycook) [inline codebasin/preprocessor.py:700]: I think this one should remain `args`: we're not talking about a command here.

File: codebasin/preprocessor.py [hunk 3]
```diff
@@ -1482,43 +1482,43 @@ class MacroFunction(Macro):
     Represents a macro function definition.
     """
 
-    def __init__(self, name, args, replacement):
-        self.args = [x.token for x in args]
+    def __init__(self, name, argv, replacement):
+        self.argv = [x.token for x in argv]
```
Comment (Pennycook) [inline codebasin/preprocessor.py:1486]: I think this one should remain `args`: we're not talking about a command here.

File: etc/coverage.py [hunk 4]
```diff
@@ -27,10 +27,10 @@
                         help="path to compilation database JSON file")
     parser.add_argument('ofile', metavar='OUTPUT',
                         help="path to coverage JSON file")
-    args = parser.parse_args()
+    argv = parser.parse_args()
```
Comment (Pennycook) [inline etc/coverage.py:30]: I think this one should remain `args`: same logic as for codebasin.py.

File: etc/preprocess.py [hunk 5]
```diff
@@ -35,7 +35,7 @@
     parser.add_argument('--summarize', dest='summarize', action='store_true', default=False,
                         help="summarize code blocks with SLOC count")
     parser.add_argument('filename', metavar='FILE', action='store')
-    args = parser.parse_args()
+    argv = parser.parse_args()
```
Comment (Pennycook) [inline etc/preprocess.py:38]: I think this one should remain `args`: same logic as for codebasin.py and coverage.py.

File: tests/macro_expansion/test_macro_expansion.py [hunk 6]
```diff
@@ -96,7 +96,7 @@ def test_variadic(self):
 
         for def_string in [
             "eprintf(...)=fprintf(stderr, __VA_ARGS__)",
-                "eprintf(args...)=fprintf(stderr, args)"]:
+                "eprintf(argv...)=fprintf(stderr, argv)"]:
```
Comment (Pennycook) [inline tests/macro_expansion/test_macro_expansion.py:99]: I think this one should remain `args`: the parameter to `eprintf` in the C documentation is called `args`.

File: tests/parsers/test_directive_parser.py [hunk 7]
```diff
@@ -16,22 +16,22 @@ def test_define(self):
         node = preprocessor.DirectiveParser(tokens).parse()
         self.assertTrue(isinstance(node, preprocessor.DefineNode))
         self.assertTrue(str(node.identifier) == "FOO")
-        self.assertTrue(node.args is None)
+        self.assertTrue(node.argv is None)
```
Comment (Pennycook) [inline tests/parsers/test_directive_parser.py:19]: I think these will need to be changed as a result of reverting `argv` to `args` in the classes themselves.

**Review by Pennycook — CHANGES_REQUESTED** [review]
Thank you again for working on this.

I didn't appreciate that we had a few different conventions for `args` here -- thank you for being thorough and looking for them all!  I think there are a few cases here where renaming things to `argv` isn't appropriate; I've convinced myself that it only makes sense to use `argv` as a name when the variable is a list of strings representing a command (like one would get from `sys.argv`).

I've gone through the changes and highlighted cases where I think we should revert to `args` in keeping with the logic above.  @laserkelvin, could you also take a quick look to see if you agree that this convention makes sense?

Code changes aside, it would be great if you could also:
- Rebase to remove the first two (unused) commits
- Amend the commit to include a sign-off (with `git commit -s`)

I suspect it would be easiest to do both of these things at the same time, by squashing everything into one (signed) commit.  You can do that without opening a new pull request, by making your changes locally and then force-pushing to the same branch.

File: codebasin.py (hunk 1 above)
Comment (laserkelvin) [reply codebasin.py:83]: Agreed, because even the `argparse` function is called `parse_args` and not `parse_argv` ;)

**Comment by itsjayway** [comment]
thank you for your feedback. I've pushed a commit that keeps the `argv` changes where they relate to commands. 

I'm not sure how I erased my commit history, but hopefully this change is appropriate

**Review by Pennycook — APPROVED** [review]

**Comment by Pennycook** [comment]
Thank you for contributing! :tada:

**Pull request merged**